File: examples/sdks/php/execute.php
<?php

declare(strict_types=1);

use IRoute\IRouteClient;

require_once __DIR__ . '/../../../sdks/php/vendor/autoload.php';

$client = new IRouteClient(
    getenv('IROUTE_URL') ?: 'http://localhost:8080',
    getenv('IROUTE_TOKEN') ?: null,
    getenv('IROUTE_TENANT') ?: 'demo',
    getenv('IROUTE_ACTOR') ?: 'sdk-example',
);
$result = $client->execute([
    'taskType' => 'email.draft',
    'input' => ['purpose' => 'Confirm the SDK quick start.'],
    'idempotencyKey' => 'php-example-' . bin2hex(random_bytes(8)),
]);
echo json_encode($result, JSON_PRETTY_PRINT | JSON_THROW_ON_ERROR) . PHP_EOL;

File: sdks/rust/examples/execute.rs
use iroute_sdk::{ClientOptions, IRouteClient};
use std::env;
use std::time::{SystemTime, UNIX_EPOCH};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = IRouteClient::new(
        env::var("IROUTE_URL").unwrap_or_else(|_| "http://localhost:8080".into()),
        ClientOptions {
            token: env::var("IROUTE_TOKEN").ok(),
            tenant_id: Some(env::var("IROUTE_TENANT").unwrap_or_else(|_| "demo".into())),
            actor_id: Some(env::var("IROUTE_ACTOR").unwrap_or_else(|_| "sdk-example".into())),
            permission_scopes: Vec::new(),
        },
    );
    let suffix = SystemTime::now().duration_since(UNIX_EPOCH)?.as_nanos();
    let request = format!(
        r#"{{"taskType":"email.draft","input":{{"purpose":"Confirm the SDK quick start."}},"idempotencyKey":"rust-example-{suffix}"}}"#
    );
    println!("{}", client.execute_json(&request, Some(&format!("rust-example-{suffix}")))?);
    Ok(())
}
